src/plugins/intel_cpu/src/nodes/matmul.cpp

```diff
@@ -624,15 +624,21 @@ void MatMul::prepareParams() {
         primitive_desc_iterator itpd = desc.createPrimitiveDescriptorIterator(engine, key.attr);
         matmul::primitive_desc prim_desc;
 
+        auto itpd_first = itpd;
         while (static_cast<bool>(itpd))  {
             impl_desc_type impl_type = parse_impl_name(itpd.impl_info_str());
 
             if (impl_type == key.implType) {
                 prim_desc = itpd.get();
                 break;
             }
-            if (!itpd.next_impl())
-                return matmul();
+            if (!itpd.next_impl()) {
+                // implType(AMX) will be not in list when input dynamic shape for AMX.
+                // Bcause AMX will be available when it assumes batch 16 for dynamic shape in compile model.
+                // However if we only pass batch 1 in runtime, AMX will be not available(Performance is low).
```

Comment:
```suggestion
                // implType(AMX) may not be in the list when the input shapes are dynamic in the case of AMX.
                // AMX is available on the compile model stage when the dummy shape is used.
                // However, if we pass batch 1 at runtime, AMX will not be available (due to the oneDNN implementation selection algorithm).
```

Reply:
My version would be 😄:
"In case of dynamic shapes an implementation type chosen as optimal for a primitive_desc with undefined input shapes, is not necessarily available for the primitive_desc with defined shape.
Example: brgemm_avx512_amx is available for a primitive with undefined input shapes but not available for primitive_desc with input batch 1"

Reply:
@EgorDuplensky gave the root cause. It will be great.